Slingshot.Api/ng/Scripts/app.js

```diff
@@ -393,10 +468,39 @@ angular.module('formApp', ['ngAnimate', 'ui.router'])
     $scope.canMoveToNextStep = function(){
         var isValid = true;
 
-        if (!$scope.formData.tenant || !$scope.formData.subscription || !$scope.formData.params) {
+        if (!$scope.formData.tenant
+            || !$scope.formData.subscription
+            || !$scope.formData.params
+            || (creatingNewRg() && !$scope.formData.existingResourceGroup.name)) {
             return false;
         }
 
+        var rgs = $scope.formData.subscription.resourceGroups;
+        if(creatingNewRg()){
+
+            var regex = new RegExp("^[0-9a-zA-Z\\()._-]+[0-9a-zA-Z()_-]$");
+            if(!regex.test($scope.formData.newResourceGroup.name)){
+                $scope.formData.resourceGroupError = "Invalid Resource Group Name";
+                isValid = false;
+            }
+            else{
+                for(var i = 0; i < rgs.length; i++){
+                    if($scope.formData.newResourceGroup.name.toLowerCase() === rgs[i].name.toLowerCase()){
+                        
+                        $scope.formData.resourceGroupError = "Resource Group Exists";
+                        isValid = false;
+                    }
+                }
+            }
+
+            if(isValid){
+                $scope.formData.resourceGroupError = null;
+            }
+        }
+        else{
+            $scope.formData.resourceGroupError = null;
```

Comment:
Actually I do want to do it like this.  The reason is because the new RG field may be valid, but if other fields are invalid, then it won't reset the RG-specific error.